### PR TITLE
[Subs 1377] Implement file content validator

### PR DESCRIPTION
**Pull request by ke4**
Add the implementation of the file content validator
Next step: add the deployment part of the build script

### CODE REVIEW

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/config/CommandLineParams.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import lombok.Data;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+
+@Data
+@Configuration
+public class CommandLineParams {
+
+    @Value("${fileContentValidator.validationResultUUID:}")
```
Comment (davidrichardson) [inline src/main/java/uk/ac/ebi/subs/filecontentvalidator/config/CommandLineParams.java]: Why do we need the ':' ?

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidationHandler.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import uk.ac.ebi.subs.validator.data.structures.ValidationAuthor;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class FileContentValidationHandler {
```
Comment (davidrichardson) [inline src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidationHandler.java:21]: This name seems a bit generic. I can't tell what role it plays.

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidationHandler.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                generateSingleValidationResultsEnvelope(singleValidationResultList);
+
+        sendValidationMessageToAggregator(singleValidationResultsEnvelope);
+    }
+
+    private SingleValidationResultsEnvelope generateSingleValidationResultsEnvelope(List<SingleValidationResult> singleValidationResults) {
+        return new SingleValidationResultsEnvelope(
+                singleValidationResults,
+                CONTENT_VALIDATION_RESULT_VERSION,
+                fileContentValidator.getCommandLineParams().getValidationResultUUID(),
```
Comment (davidrichardson) [inline src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidationHandler.java]: Why do we get the command line parameters from the validator? We could wire the dependency into this class and use it here.

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java [hunk 4]
```diff
@@ -1,60 +1,117 @@
...
-import java.util.List;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.UUID;
 
+@Data
+@Service
 public class FileContentValidator {
```
Comment (davidrichardson) [inline src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java:24]: We may as well call this class what it is - a fastq content validator

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java [hunk 5]
```diff
@@ -1,60 +1,117 @@
...
+
+    private void validateFileType() {
+        if (!FileType.forName(getCommandLineParams().getFileType())) {
+            throw new NotSupportedFileTypeException(getCommandLineParams().getFileType());
+        }
+    }
+
+    SingleValidationResult buildSingleValidationResult() {
+        SingleValidationResult singleValidationResult =
+                new SingleValidationResult(ValidationAuthor.FileContent, UUID.randomUUID().toString());
```
Comment (davidrichardson) [inline src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java]: Could you add a comment to explain why there's a random UUID here?

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileType.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package uk.ac.ebi.subs.filecontentvalidator.service;
+
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public enum FileType {
+    FASTQ, BAM, CRAM, VCF, FLATFILE;
```
Comment (davidrichardson) [inline src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileType.java]: Can we leave FLATFILE out for now, it's not on our roadmap yet?

**Review by davidrichardson — COMMENTED** [review]
I've added a few questions. This looks OK, questions are on the minor details.

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java (hunk 4 above)
Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java:24]: Yes for now, but I wanted to make it generic

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/config/CommandLineParams.java (hunk 1 above)
Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/filecontentvalidator/config/CommandLineParams.java]: I fixed it

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileType.java (hunk 6 above)
Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileType.java]: Yes. I removed it

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java (hunk 4 above)
Comment (davidrichardson) [reply src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java:24]: How about one implementation class per format?

Comment (davidrichardson) [reply src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java:24]: never mind, we can leave this till there's more than one type to validate

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileType.java (hunk 6 above)
Comment (davidrichardson) [reply src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileType.java]: 👍

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java (hunk 5 above)
Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java]: I need to fix this and replace it with the file's UUID

**Review by davidrichardson — APPROVED** [review]

**Pull request merged**